SU-SWS/decanter
# Use %placeholder and @extend on button mixins to see if it makes code less repetitive

# READY FOR REVIEW

# Summary
- We will have a discussion on using @extend on some base styles when using mixins. This is a test to see if this is a good approach and what can be done differently.
- I picked the button mixin to test because it's simple and get used by other components a lot, and it has a few variants.
- Created a `%button-base` class with the button base styles that are mostly common to all button variants, and also used by some more complex components such as the CTA (uses the `.su-button--big` variant).
- @extend this `%button-base` class in the variants' mixins and see if it reduces our output CSS file size and scss file size.

# Needed By (Date)
- Maybe take a look before Zoom meeting on Wed?

# Urgency
- N/A

# Steps to Test
1. Pull this branch and run `grunt styleguide`
2. Check in the components Button, Link, CTA section that all the buttons still look and behave the same as on the live site.
3. Look at the files changed tab on this GitHub page and see what's deleted and what's added in the output CSS and scss styles.

# Affected Projects or Products
- Decanter

# Associated Issues and/or People
- If this get merged we'll have to make sure the card and hero components stay the same or we'll have to do some small refactor.
